src/Filter/Filter.php

```diff
@@ -25,9 +27,13 @@ public function apply($query, $value)
 
         $field = $this->getParentAssociationMappings() ? $this->getName() : $this->getFieldName();
 
+        // NEXT_MAJOR: Remove null argument.
         $this->filter($query, null, $field, $value);
     }
 
+    // NEXT_MAJOR: Uncomment this code.
+    // abstract protected function filter(ProxyQueryInterface $queryBuilder, $field, $value);
```

Comment:
```suggestion
    // abstract protected function filter(ProxyQueryInterface $queryBuilder, string $field, mixed $value);
```

Reply:
I agree for string, but mixed won't be allowed before php 8 and we still support older versions.